### PR TITLE
feat: allow controlling node env config via engineconfig/cli

### PR DESCRIPTION
**Pull request by AdamYahid**

### CODE REVIEW

File: packages/engineer/src/application-proxy-service.ts [hunk 1]
```diff
@@ -86,7 +81,8 @@ export class TargetApplication extends Application {
         return super.getFeatureEnvDefinitions(features, configurations);
     }
 
-    public setNodeEnvManager(nem: NodeEnvironmentsManager) {
+    public setNodeEnvManager(nem: NodeEnvironmentsManager, nodeEnvironmentsMode?: LaunchEnvironmentMode) {
```
Comment (RomanYarik) [inline packages/engineer/src/application-proxy-service.ts:84]: no need for that.
When you do 'new NodeEnvironmentManager(..)' you set up the mode. any specific reason for saving the mode explicitly?

File: packages/engineer/test/fixtures/engine-config/feature/x.feature.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const NODE_1 = new Environment('node_1', 'node', 'single');
+export const NODE_2 = new Environment('node_2', 'node', 'single');
+export const MAIN = new Environment('main', 'window', 'single');
+
+export default new Feature({
+    id: 'XTestFeature',
+    api: {
+        node_env_1: Service.withType<{ getPid: () => number }>().defineEntity(NODE_1).allowRemoteAccess(),
+        node_env_2: Service.withType<{ getPid: () => number }>().defineEntity(NODE_2).allowRemoteAccess(),
```
Comment (RomanYarik) [inline packages/engineer/test/fixtures/engine-config/feature/x.feature.ts:11]: camelCase

**Review by RomanYarik — CHANGES_REQUESTED** [review]

File: packages/engineer/src/application-proxy-service.ts (hunk 1 above)
Comment (AdamYahid) [reply packages/engineer/src/application-proxy-service.ts:84]: You pass the mode on the request to runServerEnvironments, not to the node environment manager constructor

Comment (RomanYarik) [reply packages/engineer/src/application-proxy-service.ts:84]: Cool, my bad

**Pull request merged**